src/helpers.js

```diff
@@ -155,3 +155,99 @@ export const isRectSelected = rect => selection => {
 
   return true;
 };
+
+export const transpose = m => m[0].map((x, i) => m.map(x => x[i]));
+
+export const convertTableNodeInArrayOfRows = tableNode => {
+  const map = TableMap.get(tableNode);
+  const rows = [];
+  for (let rowIndex = 0; rowIndex < map.height; rowIndex++) {
+    const rowCells = [];
+    const seen = {};
+
+    for (let colIndex = 0; colIndex < map.width; colIndex++) {
+      const cellPos = map.map[rowIndex * map.width + colIndex];
+      const cell = tableNode.nodeAt(cellPos);
+      const rect = map.findCell(cellPos);
+      if (seen[cellPos] || rect.top !== rowIndex) {
+        rowCells.push(null);
+        continue;
+      }
+      seen[cellPos] = true;
+
+      rowCells.push(cell);
+    }
+
+    rows.push(rowCells);
+  }
+
+  return rows;
+};
+
+export const convertArrayOfRowsInTableNode = (tableNode, tableArray) => {
+  const rowsPM = [];
+  const map = TableMap.get(tableNode);
+  for (let rowIndex = 0; rowIndex < map.height; rowIndex++) {
+    const row = tableNode.child(rowIndex);
+    const rowCells = [];
+
+    for (let colIndex = 0; colIndex < map.width; colIndex++) {
+      if (!tableArray[rowIndex][colIndex]) {
+        continue;
+      }
+      const cellPos = map.map[rowIndex * map.width + colIndex];
+
+      const cell = tableArray[rowIndex][colIndex];
+      const cellData = [
+        Object.assign({}, cell.attrs),
+        cell.content,
+        cell.marks
+      ];
+
+      const oldCell = tableNode.nodeAt(cellPos);
+      const newCell = oldCell.type.createChecked.apply(oldCell.type, cellData);
+      rowCells.push(newCell);
+    }
+
+    rowsPM.push(row.type.createChecked(row.attrs, rowCells, row.marks));
+  }
+
+  const newTable = tableNode.type.createChecked(
+    tableNode.attrs,
+    rowsPM,
+    tableNode.marks
+  );
+
+  return newTable;
+};
+
+export const moveTableRowsOrColumns = (
+  table,
+  indexesOrigin,
+  indexesTarget,
+  isColumns
+) => {
+  let direction =
+    indexesOrigin[0] > indexesTarget[0] ? 'right-to-left' : 'left-to-right';
+
+  let rows = convertTableNodeInArrayOfRows(table.node);
+
+  if (isColumns) {
+    rows = transpose(rows);
+  }
+
+  const rowsExtracted = rows.splice(indexesOrigin[0], indexesOrigin.length);
```

Comment:
could you please add comments for everything starting from this line up to return? imagine someone else is looking into it for the first time with zero context and has to fix/update this logic

Reply:
Usually, we don't have comments explaining code on this repository, why should I add this now if we have tests for it?

Reply:
What I can do is add more unit tests for this function.

Reply:
either is fine, the reason I asked is because this code is hard to digest, the logic is not trivial comparing to everything else in here.

Reply:
I don't agree, there a lot of complex logic to digest on this repository, like: `getCellsInColumn`, `getCellsInRow`, `cloneRowAt`, `removeSelectedColumns` and others with zero comented lines and a not so good documentation.

Reply:
> Usually, we don't have comments explaining code on this repository, why should I add this now

There's no existing comments doesn't mean that we should never add any new.